data/src/main/java/com/microsoft/azure/kusto/data/BaseClient.java

```diff
@@ -142,8 +153,13 @@ public static DataServiceException createExceptionFromResponse(String url, HttpR
                 isPermanent);
     }
 
+    private Context getContextTimeout(long timeoutMs) {
+        int requestTimeout = timeoutMs > Integer.MAX_VALUE ? Integer.MAX_VALUE : Math.toIntExact(timeoutMs) + CLIENT_SERVER_DELTA_IN_MILLISECS;
+        return Context.NONE.addData("azure-response-timeout", Duration.ofMillis(requestTimeout));
```

Comment:
Is this documented somewhere? we might want to put a comment with the link for future reference

Reply:
not exactly but ill ref the changelog